dataframe-arrow/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/arrowReadingImpl.kt

```diff
@@ -172,6 +173,10 @@ private fun StructVector.values(range: IntRange): List<Map<String, Any?>?> = ran
     getObject(it)
 }
 
+private fun NullVector.values(range: IntRange): List<Any?> = range.map {
```

Comment:
And if I'm correct, this List will always only contain `null`s, right? Then the return type should probably be `List<Nothing?>`.
That's in line with the type of `val a = listOf(null)`.

Reply:
I tried to apply
```patch
-private fun NullVector.values(range: IntRange): List<Any?> = range.map {
-    getObject(it)
+private fun NullVector.values(range: IntRange): List<Nothing?> = range.map {
+    assert(getObject(it) == null)
+    null
 }
 ```
 
This provides build error
```
 > Task :dataframe-arrow:compileKotlin FAILED
e: file:///home/akopilov@dellin.local/workspace/kotlin/dataframe/dataframe-arrow/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/arrowReadingImpl.kt:254:51 Cannot use 'Nothing' as reified type parameter
```

Reply:
Ah yes of course, `typeOf<Nothing?>()` also doesn't work.
But there are workarounds for that, like here https://github.com/Kotlin/dataframe/blob/341365f4b986a47e8955f369582620a147370d02/core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/TypeUtils.kt#L464

So you can make a workaround that still has the `applyNullability` (with checked `NullabilityOptions`) done to the list:

```kotlin
private fun nothingType(nullable: Boolean): KType =
    if (nullable) {
        typeOf<List<Nothing?>>()
    } else {
        typeOf<List<Nothing>>()
    }.arguments.first().type!!

@JvmName("withTypeNullableNothingList")
private fun List<Nothing?>.withTypeNullable(
    expectedNulls: Boolean,
    nullabilityOptions: NullabilityOptions,
): Pair<List<Nothing?>, KType> {
    val nullable = nullabilityOptions.applyNullability(this, expectedNulls)
    return this to nothingType(nullable)
}
```

At the end it's just a small piece of help to the user: for instance, if they add all values from their `Nothing?` column and, say, a `String` column, the result should be a `List<String?>`. If the first were `Any?` then the result would be `List<Any?>` which is not helpful.

Reply:
Done (I hope that I understood correctly what do you mean)